### ISSUE TITLE
no-invalid-position-declaration on overflow-wrap inside @supports

### ISSUE THREAD

**Issue by t0byman** [stylelint/stylelint#8970]
### What minimal example or steps are needed to reproduce the bug?

```css
word-break: auto-phrase;

@supports not (word-break: auto-phrase) {
    overflow-wrap: break-word;
}
```

generates a `Unexpected invalid position declaration (no-invalid-position-declaration)` error on the `overflow-wrap` property. I think this is a bug/false positive. Should be allowed right?

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-invalid-position-declaration": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint \"src/**/*.{css,vue}\" --cache
```

### Which Stylelint-related dependencies are you using?

```json
"@stylistic/stylelint-plugin": "^4.0.1",
"stylelint-config-recommended-vue": "^1.6.1",
"stylelint-config-standard": "^40.0.0",
"stylelint": "^17.0.0",
```

### What did you expect to happen?

For it not to error / to be allowed.

### What actually happened?

```
src/styles/_tailwind/util.css
  15:9   ✖  Unexpected invalid position declaration         no-invalid-position-declaration
```

**Comment by jeddy3** [comment]
@t0byman Thanks for the report and for using the template.

The rule will flag both declarations because they are outside of a [style rule](https://drafts.csswg.org/css-syntax/#style-rule) ([Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIDuEBOUC0AjVMCGA1vAAR4CuALhBgA4AWqeAzjANwA6AdlwAJNk0aaCk2KcIFYgAoU6bLkIlyVWg2YwAlMWBdie4hABuMVADMANhCQYkjGiRz4CNtFA6cAviAA0IUwEtzGAA5PABbOEQYAA9wmiCAOjAmJh9wCE4AgHMEEBihVElYU3JzSR1OPXYQVDIgpmqSCv1iavEMf05DPHN-TCEmfwp-DIxYMHM8RmGMxuIKWphdYg8uD3c0yEz-LIAxNDC8ClymCgBPIN7OCiSM7ISwgCtU31gaVMRm1pBTi5grihzaqTCgwU7VbzLaq-S6dCgYLbZDCnPCcKBTKBAkAgsGAkBrLggDxAA)). It won't flag them inside one ([Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilB3CATlALQBGBMSA1vCkgK4AuExADgBYFIDOMA3Fiw4AAt3qtWhRtxSYIjFAAp8RMhWq0GzNpx4wAlGiE4UEAG4wCAMwA2EXMVxdWtcpSqPCUAdhQBfLADMLAB6ACpTAhQwkMFMVAxfFRI3DTomFg4uXh8grFFxSQJpWXklZLV3TQydbIMjXwTjHHNLW3tHZ1d1D2SfHCC-EAAaECsASxsYADkkAFs4RBgAD3nWKYA6MG5uEfAITAmAcwQQFaKFWCsGGwVEnHQQAnop7kfae5NHuWJxzDMkDZxiRJNxxoxxgdiLAwDYkFwIQd3ihGM8YMZBj49pBDuMjgAxQhzJCMU7cRgATymQMwjC2B2OGzmACtdqNYKxdohPo9yVSYDTGMjHnDGDByY9hsZeZTqX9GMQccdiOSkJgoPCoMKQKLxUKQIEsCA-EA)):

```css
a {
  word-break: auto-phrase;

  @supports not (word-break: auto-phrase) {
    overflow-wrap: break-word;
  }
}

/* or */

a {
  word-break: auto-phrase;
}

@supports not (word-break: auto-phrase) {
  a {
    overflow-wrap: break-word;
  }
}
```

**Comment by t0byman** [comment]
@jeddy3 

Wait I only gave you half my code, it's actually inside a tailwind 4 `@utility` directive:
```css
@utility word-break {
    hyphens: auto;
    /* word-break: break-word; is deprecated */
    word-break: auto-phrase;

    @supports not (word-break: auto-phrase) {
        overflow-wrap: break-word;
    }
}
```

`'no-invalid-position-declaration': [true, { ignoreAtRules: ['at-rules'] }]` ([Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBXALgSwDYbQTwAIB3CAJygFoAjUmAQwGsDgAdAOwM4IAs8AHbjDYBneATroIAbnZcCAegBUxMpRr0GY9YwolyUghmEFYfWmDpoYUAovmyuetbUZiJaCBQGk6wmDLYHTiRhFD4+MjRjNgg0AgAKJ2oXTXFJL24fPwBKZiC5AggANxhSADMsCCJdHz4tFN1VALkAX3YWkAAaEDLsGAA5OgBbOEQYAA9hviwYADowYWEu8Ag2XoBzBBAJiNI42DKJLDjWDgIWEFIUGeELsVO5C5iKDDYiuhxKCOFcDFWKWBgLB0HyYVZ3AgAbTQVxgnWYhnWMVoAEE0AAla4wURQi6WChXG4XAC6BBaxIcbTYLQCy0gaww6wAYmQhpYtsJ8DMcGw0PNVhtZkMAFZLbqmJaIB7nECcvDc15oCEXYFWTkXToOC5yhW8ij0jYUTl0NhQEFQZUgVXYpUgdrsEAtIA))
doesn't make the error go away. It seems 'at-rules' doesn't recognize `@utility` or `@custom-variant` only
`'no-invalid-position-declaration': [true, { ignoreAtRules: ['utility', 'custom-variant'] }],`
works. Is this an oversight?

**Comment by jeddy3** [comment]
Oh, I see. Yes, that'll flag as `@utility` isn't a standard at-rule.  You'll want to ignore it. 

The `ignoreAtRules` option ignores declarations within nested at-rules within the specified at-rules. 

So, you'll need:

```js
export default {
  "rules": {
    "no-invalid-position-declaration": [true, { ignoreAtRules: ["utility"] }]
  }
};
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBXALgSwDYbQTwAIB3CAJygFoAjUmAQwGsDgAdAOwM4IAs8AHbjDYBneATroIAbnZcCAegBUxMpRr0GY9YwolyUghmEFYfWmDpoYUAovmyuetbUZiJaCBQGk6wmDLYHTiRhFD4+MjRjNgg0AgAKJ2oXTXFJL24fPwBKZiC5AggANxhSADMsCCJdHz4tFN1VALkAX3YWkAAaEDLsGAA5OgBbOEQYAA9hviwYADowYWEu8Ag2XoBzBBAJiNI42DKJLDjWDgIWEFIUGeELsVO5C5iKDDYiuhxKCOFcDFWKWBgLB0HyYVZ3AgAbTQVxgnWYhnWMVoAEE0AAla4wURQi7obC4PAXAC6BBaxIcbTYLQCy0gaww6wAYmQhpYtsJ8DMcGw0PNVhtZkMAFZLbqmJaIB7nECcvDc15oCEXYFWTkXToOC5yhW8ij0jYUTl0NhQEFQZUgVXYpUgdrsEAtIA).

You can also extend the [`@dreamsicle.io/stylelint-config-tailwindcss`](https://github.com/dreamsicle-io/stylelint-config-tailwindcss/tree/main) config, which configures Stylelint for some of Tailwind's nonstandard constructs.

They [have one](https://github.com/dreamsicle-io/stylelint-config-tailwindcss/blob/b3a4591be93c43a3d91b927d51b7546207fa6b0a/src/stylelint.config.mjs#L50-L59) for `nesting-selector-no-missing-scoping-root` but not for `no-invalid-position-declaration` yet:

```js
	rules: {
		"import-notation": "string",
		"nesting-selector-no-missing-scoping-root": [true, {
			ignoreAtRules: [
				"custom-variant",
				"utility",
			],
		}],
	},
```

You may want to open an issue or PR there to add one for `no-invalid-position-declaration` if you end up extending that config.

**Comment by t0byman** [comment]
Ooh good tip,  thanks!

**Comment by jeddy3** [comment]
(I just noticed that the package has [stylelint 16 as a peer dep](https://github.com/dreamsicle-io/stylelint-config-tailwindcss/blob/b3a4591be93c43a3d91b927d51b7546207fa6b0a/package.json#L37). We only released Stylelint 17 earlier today, so you may need to give them a bit of time to update or help the maintainers out with a PR to add Stylelint 17 as a peer dep.)

Does the following resolve the issue for you, and can I close?:

```js
export default {
  "rules": {
    "no-invalid-position-declaration": [true, { ignoreAtRules: ["utility"] }]
  }
};
```

**Comment by t0byman** [comment]
@jeddy3 You can close, I created the PR, including the Stylelint 17 upgrade.

**Comment by theenoahmason** [comment]
Thanks for this helpful discussion and contributions @t0byman and @jeddy3 - We've implemented these notes into the latest release `1.2.2`!

I've credited you both in https://github.com/dreamsicle-io/stylelint-config-tailwindcss/releases/tag/1.2.0

- Github release: https://github.com/dreamsicle-io/stylelint-config-tailwindcss/releases/tag/1.2.2
- NPM release: https://www.npmjs.com/package/@dreamsicle.io/stylelint-config-tailwindcss/v/1.2.2
- Changes: https://github.com/dreamsicle-io/stylelint-config-tailwindcss/compare/1.1.2...1.2.2

**Comment by theenoahmason** [comment]
@jeddy3 @t0byman Just updated the peer deps to allow `^16.17.0` or `^17.0.0` and tested with both to ensure support for `17` and backwards compat with `16.17` and updated the above comment to reference `1.2.2` in case someone lands here. Thanks again for your feedback, it was helpful 👍
